lumicks/pylake/tests/test_imaging_confocal/test_scan_slice_crop.py

```diff
@@ -0,0 +1,411 @@
+        scan[Dummy(), :, :]
+
+    with pytest.raises(IndexError, match="Slicing by Dummy is not supported"):
+        scan[Dummy() : Dummy(), :, :]
+
+
+def test_crop_missing_channel(test_scan_missing_channels):
+    """Make sure that missing channels are handled appropriately when cropping"""
+
+    scan, _ = test_scan_missing_channels[("red",)]
+    np.testing.assert_equal(scan[:, 0:2, 1:3].get_image("red"), np.zeros((2, 2)))
+
+
+def test_scan_slicing_by_time(test_scan_slicing):
+    multi_frame, ref = test_scan_slicing
+    multi_frame = multi_frame[:]
+
+    rng = (np.vstack(ts := ref.timestamps.timestamp_ranges) - ts[0][0]) * 1e-9
+    wiggle = 0.01
+
+    # start of frame 2, until frame 2 signal stop
+    assert isinstance(multi_frame[f"{rng[2, 0]}s":f"{rng[2, 1]}s"], EmptyScan)
+
+    # start of frame 2, just over frame 2 signal stop
+    sliced = multi_frame[f"{rng[2, 0]}s":f"{rng[2, 1] + wiggle}s"]
+    np.testing.assert_equal(sliced.get_image("rgb"), ref.image[2])
+
+    # start of frame 2, until frame 3 signal stop
+    sliced = multi_frame[f"{rng[2, 0]}s":f"{rng[3, 1]}s"]
+    np.testing.assert_equal(sliced.get_image("rgb"), ref.image[2])
+
+    # start of frame 2, just over frame 3 signal stop
+    sliced = multi_frame[f"{rng[2, 0]}s":f"{rng[3, 1] + wiggle}s"]
+    np.testing.assert_equal(sliced.get_image("rgb"), ref.image[2:4])
+
+    # start of frame 2, until start of frame 4
+    sliced = multi_frame[f"{rng[2, 0]}s":f"{rng[4, 0]}s"]
+    np.testing.assert_equal(sliced.get_image("rgb"), ref.image[2:4])
+
+    # from start of frame 2
+    sliced = multi_frame[f"{rng[2, 0]}s":]
+    np.testing.assert_equal(sliced.get_image("rgb"), ref.image[2:])
+
+    # from just after start of frame 2
+    sliced = multi_frame[f"{rng[2, 0] + wiggle}s":]
+    np.testing.assert_equal(sliced.get_image("rgb"), ref.image[3:])
+
+    # until just after end of frame 1 signal
+    sliced = multi_frame[:f"{rng[2, 1] + wiggle}s"]
+    np.testing.assert_equal(sliced.get_image("rgb"), ref.image[:3])
+
+    # until end of frame 1 signal
+    sliced = multi_frame[:f"{rng[2, 1]}s"]
+    np.testing.assert_equal(sliced.get_image("rgb"), ref.image[:2])
+
+
+def test_scan_slicing_by_negative_time(test_scan_slicing):
+    multi_frame, ref = test_scan_slicing
+    multi_frame = multi_frame[:]
+
+    rng = (np.vstack(ts := ref.timestamps.timestamp_ranges) - ts[0][0]) * 1e-9
+    rng_dead = (np.vstack(ts := ref.timestamps.timestamp_ranges_deadtime) - ts[0][0]) * 1e-9
+
+    scan_time = rng[0, 1] - rng[0, 0]
+    dead_time = rng_dead[0, 1] - rng[0, 1]
+    frame_time = scan_time + dead_time
+    wiggle = 0.01
+
+    # Note that the from-end tests are different than the ones on ImageStack because the mock
+    # scan has the dead time half on the end of this frame, and half before this frame.
+    # The stop time of this scan is at 10 seconds, the last frame runs from 9 to 9.8 seconds, this
+    # means that to chop off the last two, we need to go -1.2 from end.
```

Comment:
This comment looks out of date. Would be nice to express this in `scan_time`, `dead_time`, `frame_time` and `wiggle`.

I also don't really think the comparison to `ImageStack` makes much sense anymore unless you mention specifically where the comparable test can be found. I would probably just leave that part out.